src/app/suggestion.service.spec.ts

```diff
@@ -0,0 +1,154 @@
+
+    it("Kombination aus Buch, Liednummer und Name", () => {
+      const results = search("Iwdd! 180 Golgatha")
+      const expected = [iwdd(180)]
+      expect(results).toEqual(expected)
+    })
+
+  })
+
+  describe("Spezielles Suchverhalten", () => {
+
+    it("Groß-Kleinschreibung wird ignoriert", () => {
+      expect(search("g463")).toEqual(search("G463"))
+      expect(search("Iwdd!")).toEqual(search("IWDD!"))
+      expect(search("Golgatha")).toEqual(search("golGATHA"))
+      expect(search("Iwdd! 180 Golgatha")).toEqual(search("iwdd! 180 golgatha"))
+    })
+
+    it("Unnötigen Leerzeichen werden ignoriert", () => {
+      expect(search("Iwdd! 180 Golgatha")).toEqual(search(" iwdd!    180 golgatha  "))
+    })
+
+    it("Teilweise Treffer werden aufgelistet", () => {
+      expect(search("g46").length).toEqual(5)
+    })
+
+    it("Genaue Treffer stehen oben", () => {
+      const results = search("g46")
+      expect(results.length).toBeGreaterThan(0)
+      expect(results[0]).toEqual(iwdd(46))
+    })
+
+    it("Suchergebnisse sind alphabetisch nach Buch sortiert", () => {
+      const results = search("365")
+      expect(results.length).toEqual(2)
+      expect(results[0]).toEqual(loben(365))
+      expect(results[0]).toEqual(iwdd(365))
+    })
+
+    it("Suchergebnisse sind nach fortlaufender Liednummer sortiert", () => {
+      const results = search("65")
+      expect(results.length).toBeGreaterThan(1)
+      expect(results[0].number).toBeLessThan(results[1].number)
+    })
+
+  })
+
+  describe("Die Suche nach Qualifier wird bevorzugt, außer wenn...", () => {
+
+    it("...wird bevorzugt", () => {
+      const results = search("b")
+      const expected = [
+        loben("Ein neuer Tag beginnt"),
+      ]
+      const notExpected = [
+        iwdd("Großer Gott, wir loben dich"),
+      ]
+      expect(results).toEqual(jasmine.arrayContaining(expected))
+      expect(results).not.toEqual(jasmine.arrayContaining(notExpected))
+    })
+
+    it("...das erste Input-Zeichen keinem Qualifier entspricht (z.B. nicht 'g' oder 'b' sondern 'd' oder '3')", () => {
+      const results = search("d")
+      const expected = [
+        loben("All die Fülle ist in dir"),
+        iwdd("Großer Gott, wir loben dich"),
+      ]
+      expect(results).toEqual(jasmine.arrayContaining(expected))
+    })
+
+    it("...das zweite Zeichen keiner Zahl entspricht (z.B. 'bT')", () => {
+      const results = search("bt")
+      const expected = [
+        loben("Gott gibt Mut zum Leben"),
+        iwdd("Steht auf und lobt unsern Gott"),
+      ]
+      const notExpected = [
+        iwdd("Holy, holy, holy! Lord God Almighty"),
+      ]
+      expect(results).toEqual(jasmine.arrayContaining(expected))
+      expect(results).not.toEqual(jasmine.arrayContaining(notExpected))
+    })
+
+    it("...spätestens das dritte vierte oder x-te Zeichen wieder keine Zahl ist (z.B. 'b34g' oder 'g104&')", () => {
+      // Ich glaube, den Fall gibt's nicht.
+    })
```

Comment:
falls '(' ein qualifier werden sollte für loben (statt 'b'), dann testen:
const results = search("(2 ")
const expected = [
        loben("Du gibst das Leben (2 Liedsätze)"),
]
const not expected = [
        loben("Lobe den Herrn, meine Seele (II)")
]